ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/MixedHivePartitionPlan.java

```diff
@@ -162,12 +162,19 @@ protected boolean isFragmentFile(DataFile dataFile) {
       }
     }
 
+    @Override
+    protected boolean isUndersizedSegmentFile(DataFile dataFile) {
+      return !inHiveLocation(dataFile) && super.isUndersizedSegmentFile(dataFile);
+    }
+
     @Override
     public boolean isFullNecessary() {
       if (!reachFullInterval() && !reachHiveRefreshInterval()) {
         return false;
       }
-      return fragmentFileCount > getBaseSplitCount() || hasNewHiveData();
+      return fragmentFileCount > getBaseSplitCount()
+          || rewriteSegmentFileCount + undersizedSegmentFileCount > getBaseSplitCount()
```

Comment:
This newly added condition is very easy to trigger because for full optimizing, almost all files will be involved, and rewriteSegmentFileCount is easily exceeded by the base split count. This will result in repeated and ineffective full optimizing after full optimizing, leading to resource wastage. I suggest removing this condition and sticking to the original code in this PR.

Reply:
Done.